ide/coqide.ml

```diff
@@ -1046,16 +1046,24 @@ let build_ui () =
       ~accel:"F1"
       ~callback:(cb_on_current_term MiscMenu.show_hide_query_pane);
     GAction.group_radio_actions
-      ~callback:begin function
-        | 0 -> List.iter (fun o -> Opt.set o "off") Opt.diff_item.Opt.opts
-        | 1 -> List.iter (fun o -> Opt.set o "on") Opt.diff_item.Opt.opts
-        | 2 -> List.iter (fun o -> Opt.set o "removed") Opt.diff_item.Opt.opts
-        | _ -> assert false
+      ~init_value:(
+        let v = diffs#get in
```

Comment:
I don't think this code is run at the right time. The initial value is taken when the module is linked, and loading the preferences has not occurred yet at that point. Rather, you should tie the value of this menu to the diff preference by using its connect signal so that they are always synchronized.

Reply:
The current code works correctly.  This code is inside the build_ui function, which is executed **after** the object files are loaded.  coqide_main.ml4 ends with:

```
let () =
  load_prefs ();
  :
```

Which would be executed at object file load time.

I'm not seeing how I could do this with connect.  I expect I need code something like this:

```
rb_unset = radio "Unset diff" 0 ~label:"_Don't show diffs";
rb_unset#connect#activate ~callback:(... diffs#set "off");
rb_unset#connect#SOMETHING ~callback:(... return diffs#get)
```

In which the radio button calls the SOMETHING callback to use the preference value when it is bringing up the menu.  I've spent a good while looking at gAction.ml in the GTK library but I don't see anything candidate for SOMETHING.  (BTW, the experimentation goes very slow because it takes about an hour for me to rebuild CoqIDE on Windows--I'm stuck doing full builds.)

After playing with that a couple hours I also tried this:

```
        let active = diffs#get in
        let to_str v = if v = 1 then "on" else if v = 2 then "removed" else "off" in
        let rb1 = radio "Unset diff" 0 ~active ~label:"_Don't show diffs" in
```

but the compiler objects--there is no `~active` on `radio` (= GAction.add_radio_action).

If you can advise me some on how to get connect to work, I'll try it, but considering that this code works correctly, how about if we let it be?